core/src/main/java/org/apache/calcite/rel/rules/FilterProjectTransposeRule.java

```diff
@@ -175,11 +176,13 @@ protected FilterProjectTransposeRule(
       final RelNode input = project.getInput();
       final RelTraitSet traitSet = filter.getTraitSet()
           .replaceIfs(RelCollationTraitDef.INSTANCE,
-              () -> Collections.singletonList(
-                      input.getTraitSet().getTrait(RelCollationTraitDef.INSTANCE)))
+              () -> input.getTraitSet().getTraits(RelCollationTraitDef.INSTANCE))
```

Comment:
is this change related to this issue?

Reply:
Yes, I'm not entirely sure why; I made some adjustments based on the error message. I will revert the changes to RelDistribution later.
```
> Error while executing command ExplainCommand [sql: with t (a, b) as (select * from (values (60, 'b')))
> select * from t where a in (select deptno from "scott".dept)]
> java.sql.SQLException: Error while executing SQL "explain plan for with t (a, b) as (select * from (values (60, 'b')))
> select * from t where a in (select deptno from "scott".dept)": class org.apache.calcite.plan.RelCompositeTrait cannot be cast to class org.apache.calcite.rel.RelCollation (org.apache.calcite.plan.RelCompositeTrait and org.apache.calcite.rel.RelCollation are in unnamed module of loader 'app')
```